routers/api/v1/repo/issue.go

```diff
@@ -22,6 +23,123 @@ import (
 	milestone_service "code.gitea.io/gitea/services/milestone"
 )
 
+// SearchIssues searches for issues across the repositories that the user has access to
+func SearchIssues(ctx *context.APIContext) {
+	// swagger:operation GET /repos/issues/search issue issueSearchIssues
+	// ---
+	// summary: Search for issues across the repositories that the user has access to
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: state
+	//   in: query
+	//   description: whether issue is open or closed
+	//   type: string
+	// - name: labels
+	//   in: query
+	//   description: comma separated list of labels. Fetch only issues that have any of this labels. Non existent labels are discarded
+	//   type: string
+	// - name: page
+	//   in: query
+	//   description: page number of requested issues
+	//   type: integer
+	// - name: q
+	//   in: query
+	//   description: search string
+	//   type: string
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/IssueList"
+
+	// find repos user can access
+	repoIDs := make([]int64, 0)
+	issueCount := 0
+	for page := 1; ; page++ {
+		repos, count, err := models.SearchRepositoryByName(&models.SearchRepoOptions{
+			Page:        page,
+			PageSize:    models.RepositoryListDefaultPageSize,
+			Private:     true,
+			Keyword:     "",
+			OwnerID:     ctx.User.ID,
+			TopicOnly:   false,
+			Collaborate: util.OptionalBoolNone,
+			UserIsAdmin: ctx.IsUserSiteAdmin(),
+			UserID:      ctx.Data["SignedUserID"].(int64),
+			OrderBy:     models.SearchOrderByID,
+		})
+		if err != nil {
+			ctx.Error(500, "SearchRepositoryByName", err)
+			return
+		}
+
+		if len(repos) == 0 {
+			break
+		}
+		log.Trace("Processing next %d repos of %d", len(repos), count)
+		for _, repo := range repos {
+			issueCount += repo.NumIssues
+			repoIDs = append(repoIDs, repo.ID)
+		}
+	}
+
+	var isClosed util.OptionalBool
+	switch ctx.Query("state") {
+	case "closed":
+		isClosed = util.OptionalBoolTrue
+	case "all":
+		isClosed = util.OptionalBoolNone
+	default:
+		isClosed = util.OptionalBoolFalse
+	}
+
+	var issues []*models.Issue
+
+	keyword := strings.Trim(ctx.Query("q"), " ")
+	if strings.IndexByte(keyword, 0) >= 0 {
+		keyword = ""
+	}
+	var issueIDs []int64
+	var labelIDs []int64
+	var err error
+	if len(keyword) > 0 && len(repoIDs) > 0 {
+		issueIDs, err = issue_indexer.SearchIssuesByKeyword(repoIDs, keyword)
+	}
+
+	if splitted := strings.Split(ctx.Query("labels"), ","); len(splitted) > 0 {
```

Comment:
This condition will always be true. `strings.Split()` will return `nil` only if both `s` and `sep` are empty.
https://golang.org/pkg/strings/#Split

Reply:
I'm not sure why it would always return true as I was thinking the condition is based on the length of the slice produced from the split, but I broke it up to make it more clear in case I'm not understanding.